pkg/distributor/distributor.go

```diff
@@ -241,18 +253,41 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 		return nil, err
 	}
 
+	d.haFlag = true
```

Comment:
Can you add this to the `Config` struct and register it as a flag appropriately.